daxa/mission/erosita.py

```diff
@@ -0,0 +1,703 @@
+                    # The path to the obs_id directory
+                    obs_dir = os.path.join(self.raw_data_path, '{o}'.format(o=o))
+                    # The field the obs_id was downloaded with
+                    field_name = field_dict[o]
+                    # The path to where the obs_id was initially downloaded
+                    field_dir = os.path.join(self.raw_data_path, "temp_download", field_name)
+                    # Not including hidden files in this list
+                    all_files = [f for f in os.listdir(field_dir) if not f.startswith('.')]
+                    # Some fields are in a folder, some are just the files not in a folder
+                    # If they are in a folder, there will only be one file in all files
+                    if len(all_files) == 1:
+                        second_field_dir = all_files[0]
+                        # redefining all_files so it lists the files in the folder
+                        all_files = os.listdir(os.path.join(field_dir, second_field_dir))
+                        # redefining field_dir so in the later block, the source is correct
+                        field_dir = os.path.join(field_dir, second_field_dir)
+                        
+                    # Selecting the eventlist for the obs_id
+                    obs_file_name =  [obs_file for obs_file in all_files if o in obs_file and "eRO" not in obs_file ][0]
+                    source = os.path.join(field_dir, obs_file_name)
+                    dest = os.path.join(obs_dir, obs_file_name)
+                    shutil.move(source, dest)
+
+            # Deleting temp_download directory containing the extra files that were not the obs_id eventlists
+            shutil.rmtree(os.path.join(self.raw_data_path, "temp_download"))
+        
+        else:
+            pass
+    
+    def _get_evlist_path_from_obs(self, obs: str):
+        '''
+        Internal method to get the unfiltered, downloaded event list path for a given
+        obs id, for use in the download method. 
+
+        :param str obs: The obs id for the event list required.
+        :return: The path of the event list.
+        :rtype: str
+        '''
+        all_files = os.listdir(os.path.join(self.raw_data_path + '{o}'.format(o=obs)))
+
+         # This directory could have instrument filtered files in as well as the eventlist
+         #  so selecting the eventlist by chosing the one with 4 hyphens in
+        file_name = [file for file in all_files if len(re.findall('_', file)) == 4][0]
+
+        ev_list_path = os.path.join(self.raw_data_path + '{o}'.format(o=obs), file_name)
+
+        return ev_list_path
+    
+    def download(self, num_cores: int = NUM_CORES):
+        """
+        A method to acquire and download the eROSITA CalPV data that have not been filtered out (if a filter
+        has been applied, otherwise all data will be downloaded). Fields (or field types), which is set either 
```

Comment:
It could be tired David brain but I'm not sure the Fields sentence actually makes sense